influxdb3_server/src/http/v1.rs

```diff
@@ -106,6 +114,67 @@ impl QueryParams {
     }
 }
 
+/// Enum representing the query format for the v1/query API.
+///
+/// The original API supports CSV, JSON, and "pretty" JSON formats.
+#[derive(Debug, Deserialize, Clone, Copy)]
+#[serde(rename_all = "snake_case")]
+pub(crate) enum QueryFormat {
+    Csv,
+    Pretty,
+    Json,
+}
+
+impl QueryFormat {
+    /// Returns the content type as a string slice for the query format.
+    ///
+    /// Maps the `QueryFormat` variants to their corresponding MIME types as strings.
+    /// This is useful for setting the `Content-Type` header in HTTP responses.
+    fn as_content_type(&self) -> &str {
+        match self {
+            Self::Csv => "application/csv",
+            Self::Json | Self::Pretty => "application/json",
+        }
+    }
+
+    /// Checks if the query format is 'Pretty'.
+    ///
+    /// Determines if the `QueryFormat` is `Pretty`, which indicates that the JSON
+    /// output should be formatted in a human-readable way. Returns `true` if the
+    /// format is `Pretty`, otherwise returns `false`.
+    fn is_pretty(&self) -> bool {
+        match self {
+            Self::Csv | Self::Json => false,
+            Self::Pretty => true,
+        }
+    }
+
+    /// Extracts the [`QueryFormat`] from an HTTP [`Request`].
+    ///
+    /// Parses the HTTP request to determine the desired query format. The `pretty`
+    /// parameter indicates if the pretty format is requested via a query parameter.
+    /// The function inspects the `Accept` header of the request to determine the
+    /// format, defaulting to JSON if no specific format is requested. If the format
+    /// is invalid or non-UTF8, an error is returned.
+    fn from_request(req: &Request<Body>, pretty: bool) -> Result<Self> {
+        // In V1 pretty format is a Query params
+        if pretty {
+            return Ok(Self::Pretty);
+        }
```

Comment:
I tried out the `v1` API locally, and found that the order should be flipped such that the header takes priority. For example, if the header `Accept: application/csv` is passed, then the `pretty` query parameter is ignored.

So basically, this just needs to move down to the match arm for JSON.

Reply:
Fixed, WDYT?